src/reactUtils.js

```diff
@@ -53,34 +53,57 @@ export function useDebugAssertConstant<T>(value: T) {
 }
 
 /**
- * True just when `value` has been true continuously for the past `duration`.
+ * True just when a given state/prop/etc. has not changed for the past
+ *   `duration`.
```

Comment:
> The point is really about the meaning of "changed", more than about the nature of the `value` parameter

I think there's a point about the meaning of "changed", and also a point about the nature of the `value` parameter. (I don't think my revision captured either point very well, and perhaps neither point needs a special mention, I'm not sure. 🙂)

- As you've said, if `value` itself is mutable, then this doesn't detect any mutations within it.
- If `value` is derived from something mutable (e.g., the value `true` from `.current` on a boolean ref), and that thing mutates such that the caller would pass something different for `value` than it did last time, then this _might_ detect the change. In particular, it will detect the change if the Hook is actually called with the new `value`. That might be on the same render as the change, or on a later render, or never (e.g., before the next render, the component unmounts or the mutation is reversed). So passing a `value` with this nature is very risky, and if a caller passed a `myRef(…).current` as `value`, I'd count it as a code smell whether or not they somehow got results they were satisfied with.

Reply:
> … and that thing mutates such that the caller would pass something different (non-`===`) for `value` than it did last time, then this _might_ detect the change. In particular, it will detect the change if the Hook is actually called with the new `value`.

I see. The point you're making here is that this Hook only notices a change when the Hook is actually called, i.e. during a render -- it doesn't set up any kind of subscription to arrange to notice when something changes in between renders. (In contrast with `useState`, or react-redux's `useSelector`.)

Instead, the caller needs to arrange that. Which isn't hard -- it basically comes naturally if the caller follows normal React Hooks idioms. The value should come from something like a prop or `useState` state or `useSelector` result; and those should be immutable so that any updates deep inside them are reflected by the top-level object getting replaced with a different object, one that's `!==` from the old one. The main risk is probably from overusing `useRef`.

I think that caveat is worth mentioning in the jsdoc here, particularly because this Hook *does* arrange to notice (and cause a rerender on its own initiative) in a different case, namely when the given time elapses. I'll make a go at it and push another commit to the end of the PR branch.

Reply:
> I'll make a go at it and push another commit to the end of the PR branch.

86ea850e8

Reply:
Yeah exactly; thanks!